core/src/test/scala/org/apache/spark/ShuffleSuite.scala

```diff
@@ -271,25 +271,49 @@ abstract class ShuffleSuite extends SparkFunSuite with Matchers with LocalSparkC
   test("[SPARK-4085] rerun map stage if reduce stage cannot find its local shuffle file") {
     val myConf = conf.clone().set("spark.test.noStageRetry", "false")
     sc = new SparkContext("local", "test", myConf)
-    val rdd = sc.parallelize(1 to 10, 2).map((_, 1)).reduceByKey(_ + _)
+    val rdd = sc.parallelize(1 to 10, 1).map((_, 1)).reduceByKey(_ + _)
```

Comment:
@srowen 
I think it over access to shuffle the data file with the index file closer to the test target. so modify it.